stdlib/public/runtime/EnvironmentVariables.cpp

```diff
@@ -0,0 +1,165 @@
+//===--- CompatibiltyOverride.h - Debug variables. --------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// Debug behavior conditionally enabled using environment variables.
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/Runtime/Debug.h"
+#include "swift/Runtime/EnvironmentVariables.h"
+
+#include <errno.h>
+
+using namespace swift;
+
+// Require all environment variable names to start with SWIFT_
+static constexpr bool hasSwiftPrefix(const char *str) {
+  const char prefix[] = "SWIFT_";
+  for (unsigned i = 0; i < sizeof(prefix) - 1; i++)
+    if (str[i] != prefix[i])
+      return false;
+  return true;
+}
+#define VARIABLE(name, type, defaultValue, help) \
+  static_assert(hasSwiftPrefix(#name), "Names must start with SWIFT");
+#include "EnvironmentVariables.def"
+
+// Define backing variables.
+#define VARIABLE(name, type, defaultValue, help) \
+  type Environment::name ## _variable;
```

Comment:
Should this not be `static type ...`?

Reply:
No, this and `initializeToken` get exposed to other files through the inline getter functions in the header, so they need to be non-`static`.